network.koesling.shm-modbus.yml

```
@@ -0,0 +1,104 @@
+id: network.koesling.shm-modbus
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+command: shm-modbus
+finish-args:
+  - --device=shm
+  - --share=network
+modules:
+  - name: Modbus_TCP_client_shm
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DCLANG_FORMAT=OFF
+      - -DCOMPILER_WARNINGS=OFF
+
+    sources:
+      - type: git
+        tag: v1.2.0
+        commit: 26a61f7df3ea3190eccef224d34475278984e7db
+        url: https://github.com/NikolasK-source/modbus_tcp_client_shm.git
+
+  - name: Modbus_RTU_client_shm
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DCLANG_FORMAT=OFF
+      - -DCOMPILER_WARNINGS=OFF
+
+    sources:
+      - type: git
+        tag: v0.2.0
+        commit: 5aa84184b9dbbb4d644686126b6d03b5f70bbb0c 
+        url: https://github.com/NikolasK-source/modbus_rtu_client_shm.git
+
+  - name: write_shm
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DCLANG_FORMAT=OFF
+      - -DCOMPILER_WARNINGS=OFF
+
+    sources:
+      - type: git
+        tag: v0.1.3
+        commit: bc58b820a79fd00a699835acbfb23f8870c7a528 
+        url: https://github.com/NikolasK-source/write_shm.git
+
+  - name: SHM_random
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DCLANG_FORMAT=OFF
+      - -DCOMPILER_WARNINGS=OFF
+
+    sources:
+      - type: git
+        tag: v1.0.4
+        commit: 0920a8a36b57ac1c86483dd11ef37c98a9c6aa0a 
+        url: https://github.com/NikolasK-source/shared_mem_random.git
+
+  - name: dump_shm
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DCLANG_FORMAT=OFF
+      - -DCOMPILER_WARNINGS=OFF
+
+    sources:
+      - type: git
+        tag: v1.0.2
+        commit: b39922ac9dbb3b37a73ffc0eed3c56582d0c50e9 
+        url: https://github.com/NikolasK-source/dump_shm.git
+
+  - name: stdin-to-modbus-shm
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DCLANG_FORMAT=OFF
+      - -DCOMPILER_WARNINGS=OFF
+
+    sources:
+      - type: git
+        tag: v1.0.4
+        commit: b0b427494335e7102ea937de922ec377da77d0dd 
+        url: https://github.com/NikolasK-source/stdin_to_modbus_shm.git
+
+  - name: launch_scripts_and_metadata
+    buildsystem: simple
+    build-commands:
+      - install -D -m 755 shm-modbus ${FLATPAK_DEST}/bin/shm-modbus
+      - install -D -m 644 guided-commands.sh ${FLATPAK_DEST}/bin/guided-commands.sh
+      - install -D -m 644 manual-commands.sh ${FLATPAK_DEST}/bin/manual-commands.sh
```

```suggestion
      - install -D -m755 guided-commands.sh ${FLATPAK_DEST}/bin/guided-commands.sh
      - install -D -m755 manual-commands.sh ${FLATPAK_DEST}/bin/manual-commands.sh
```

These are executable, aren't they?

---

They are only used by shm-modbus, but never executed themselves. So they do not have to be executable.